### PR TITLE
fix(aio): close API filter selects when clicking outside (and improve a11y)

### PR DESCRIPTION
**Pull request by gkalpak**
There weren't any tests for this functionality before, but adding some is probably a good idea (if we decide to go with this approach).

An alternative approach is to bring in material's `MdSelect` (with its dependencies).

### CODE REVIEW

File: aio/src/app/embedded/api/api-list.component.ts [hunk 1]
```diff
@@ -69,9 +74,16 @@ export class ApiListComponent implements OnInit {
 
   constructor(
     private apiService: ApiService,
+    @Inject(DOCUMENT) private doc: Document,
     private locationService: LocationService) { }
 
   ngOnInit() {
+    fromEvent(this.doc, 'click')
```
Comment (petebacondarwin) [inline aio/src/app/embedded/api/api-list.component.ts]: why use this rather than `@hostBinding` on the document?

Comment (gkalpak) [inline aio/src/app/embedded/api/api-list.component.ts]: `@HostListener` is probably better. This was just a quick POC that I put together quickly, so not merge-ready.
In the meantime, we have decided to move in another direction (using material's `mdSelect` etc), so this is obsolete.

**Comment by petebacondarwin** [comment]
Please consider this PR instead #17236

**Comment by gkalpak** [comment]
Closing in favor of #17236.

**Pull request closed without merging**